From dd2797f73b078daec179a5698b5a6e5f85aa832a Mon Sep 17 00:00:00 2001
From: Behan Webster <behanw@converseincode.com>
Date: Tue, 3 Sep 2013 22:27:26 -0400
Subject: [PATCH] Kbuild, arm64, LLVMLinux: Add Kbuild support for building
 arch arm64 with Clang

Protect more options for arm64 with cc-option so that we don't get errors when
using clang instead of gcc.  Add more or different options when using clang as
well.

Author: Behan Webster <behanw@converseincode.com>
Signed-off-by: Behan Webster <behanw@converseincode.com>
Signed-off-by: Mark Charlebois <charlebm@gmail.com>
---
 arch/arm64/Makefile | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/Makefile b/arch/arm64/Makefile
index 2df5e5d..48f8fbe 100644
--- a/arch/arm64/Makefile
+++ b/arch/arm64/Makefile
@@ -19,7 +19,13 @@ LIBGCC 		:= $(shell $(CC) $(KBUILD_CFLAGS) -print-libgcc-file-name)
 
 KBUILD_DEFCONFIG := defconfig
 
-KBUILD_CFLAGS	+= -mgeneral-regs-only
+ifeq ($(COMPILER),clang)
+KBUILD_CFLAGS   += -fno-builtin,)
+KBUILD_CFLAGS   += -Wno-asm-operand-widths
+KBUILD_CFLAGS   += -Wno-initializer-overrides
+endif
+
+KBUILD_CFLAGS	+= $(call cc-option,-mgeneral-regs-only,)
 ifeq ($(CONFIG_CPU_BIG_ENDIAN), y)
 KBUILD_CPPFLAGS	+= -mbig-endian
 AS		+= -EB
-- 
1.9.1

